Source/Fuse.Common/Tests/Fuse.Common.Test/UnoEnumerable.Test.uno

```diff
@@ -0,0 +1,103 @@
+using Uno;
+using Uno.Collections;
+using Uno.Testing;
+
+using Fuse.Internal;
+
+using FuseTest;
+
+namespace Fuse.Test
+{
+	class SomeEnumerable<T> : IEnumerable<T> where T : class
+	{
+		public IEnumerator<T> GetEnumerator()
+		{
+			return new Enumerator(this);
+		}
+
+		public int EnumCount;
+		
+		class Enumerator : IEnumerator<T>, IDisposable
+		{
+			int _steps = 5;
+			SomeEnumerable<T> _source;
+			public Enumerator(SomeEnumerable<T> source)
+			{
+				_source = source;
+				_source.EnumCount++;
+			}
+			
+			public bool MoveNext() { 
+				return --_steps > 0; 
+			}
+			public T Current { get { return null; } }
+			public void Reset() { } 
+			
+			public void Dispose()
+			{
+				_source.EnumCount--;
+			}
+		}
+	}
+	
+	public class UnoEnumerableTest : TestBase
+	{
+		[Test]
+		[Ignore("https://github.com/fusetools/uno/issues/1148")]
+		public void ForeachDispose()
+		{
+			var q = new SomeEnumerable<string>();
+			foreach (var s in q) { }
+			Assert.AreEqual(0, q.EnumCount);
+		}
```

Comment:
Should I merge them with the Ignore to Uno?   I wrote them here originally since this was critically important to the `ObjectList`.